pkg/agent/api_docs.go

```diff
@@ -269,7 +269,12 @@ func newOpenAPI(ctx context.Context, docBytes []byte, baseURL string, prefix str
 		dlog.Debugf(ctx, "could not parse URL %q", baseURL)
 	} else {
 		if prefix != "" {
-			if existingPrefix != "" && keepExistingPrefix {
+
+			if existingPrefix != "" &&
+				keepExistingPrefix &&
+				strings.TrimRight(prefix, "/") != strings.TrimRight(existingPrefix, "/") &&
```

Comment:
Logically it seems correct but it's a little tough to read. Maybe we could break the `strings.TrimRight` pieces into variables so that the code isn't duplicated?